tests/framework/prototype.py

```diff
@@ -368,6 +380,76 @@ def clean_up(self):
             show_cluster_info(self.namespace)
             raise Exception("RayServiceAddCREvent clean_up() timeout")
 
+class RayJobAddCREvent(CREvent):
+    """CREvent for RayJob addition"""
+    def wait(self):
+        """Wait for RayJob to converge"""
+        start_time = time.time()
+        expected_head_pods = get_expected_head_pods(self.custom_resource_object)
+        expected_worker_pods = get_expected_worker_pods(self.custom_resource_object)
+        # Wait until:
+        #   (1) The number of head pods and worker pods are as expected.
+        #   (2) All head pods and worker pods are "Running".
+        #   (3) RayJob named "rayjob-sample" has status "SUCCEEDED".
+        converge = False
+        k8s_v1_api = K8S_CLUSTER_MANAGER.k8s_client_dict[CONST.K8S_V1_CLIENT_KEY]
+        for _ in range(self.timeout):
+            headpods = k8s_v1_api.list_namespaced_pod(
+                namespace = self.namespace, label_selector='ray.io/node-type=head')
+            workerpods = k8s_v1_api.list_namespaced_pod(
+                namespace = self.namespace, label_selector='ray.io/node-type=worker')
+            rayjob = get_custom_object(CONST.RAY_JOB_CRD, self.namespace,
+                self.custom_resource_object['metadata']['name'])
+
+            if (len(headpods.items) == expected_head_pods
+                    and len(workerpods.items) == expected_worker_pods
+                    and check_pod_running(headpods.items) and check_pod_running(workerpods.items) 
+                    and rayjob.get('status') is not None
+                    and rayjob.get('status').get('jobStatus') == "SUCCEEDED"):              
+                converge = True
+                logger.info("--- RayJobAddCREvent %s seconds ---", time.time() - start_time)
+                break
+            time.sleep(1)
+
+        if not converge:
+            logger.info("RayJobAddCREvent wait() failed to converge in %d seconds.",
+                self.timeout)
+            logger.info("expected_head_pods: %d, expected_worker_pods: %d",
+                expected_head_pods, expected_worker_pods)
+            logger.info("rayjob: %s", rayjob)
+            
+            show_cluster_info(self.namespace)
+            raise Exception("RayJobAddCREvent wait() timeout")
+
+    def clean_up(self):
```

Comment:
Will RayJob automatically clean up the Ray Pods after it succeeds?

Reply:
By default, the terminate cluster on completion flag is set to False, though we may change this behavior in the future.  (I'm not sure it's intentional that the default is False, and it doesn't seem to be documented at the moment)